numbat/src/datetime.rs

```diff
@@ -1,82 +1,65 @@
-use chrono::{DateTime, Datelike, FixedOffset, LocalResult};
-use chrono_tz::Tz;
+use jiff::{civil::DateTime, fmt::rfc2822, tz::TimeZone, Zoned};
 
-#[cfg(feature = "local-timezone")]
-pub fn get_local_timezone() -> Option<Tz> {
-    let tz_str = iana_time_zone::get_timezone().ok()?;
-    tz_str.parse().ok()
+pub fn get_local_timezone_or_utc() -> TimeZone {
+    TimeZone::system()
 }
 
-#[cfg(feature = "local-timezone")]
-pub fn get_local_timezone_or_utc() -> Tz {
-    get_local_timezone().unwrap_or(chrono_tz::UTC)
-}
-
-#[cfg(not(feature = "local-timezone"))]
-pub fn get_local_timezone_or_utc() -> Tz {
-    chrono_tz::UTC
-}
-
-pub fn parse_datetime(input: &str) -> Option<DateTime<FixedOffset>> {
-    if let Ok(dt) = chrono::DateTime::parse_from_rfc3339(input) {
-        Some(dt)
-    } else if let Ok(dt) = chrono::DateTime::parse_from_rfc2822(input) {
+pub fn parse_datetime(input: &str) -> Option<Zoned> {
+    if let Ok(dt) = DateTime::strptime("%Y-%m-%dT%H:%M:%SZ", input) {
+        // RFC 3339
+        Some(dt.to_zoned(TimeZone::UTC).unwrap()) // TODO
```

Comment:
RFC 3339 parsing could previously be done with `parse_from rfc3339` in chrono. I used `strptime` with a fixed format string for now (instead of `input.parse()`), since the `FromStr` instance on `Timestamp` seems to support a couple of other formats as well (https://docs.rs/jiff/0.1.1/jiff/fmt/temporal/index.html). And I'm not sure if it conflicts with what we have below.

Reply:
I think it is generally fine to think of `jiff::fmt::temporal` as "RFC 3339 support." That is, if you need to support RFC 3339, then that's what you should use. Now, if for some reason you actually need _strict_ RFC 3339 checking (as in, do not support anything beyond it), then Jiff doesn't really support that, other than using the `strptime` APIs as you've done here.

The stuff that it supports beyond RFC 3339 come from RFC 9557 (time zone identifiers) and (parts of) ISO 8601. For example, `2024-07-27` is supported, although you can't parse a `Zoned` from it. Here are some others:

```
2024-07-27
2024-07-27 13:30
2024-07-27 13
2024-07-27T13
2024-07-27T13:30:29.123456789
2024-07-27 13[America/New_York]
2024-07-27 13-04[America/New_York]
2024-07-27 13-04:00[America/New_York]
```

Generally it's pretty flexible. But won't support things like "use slashes instead of dashes" as a delimiter.

The other main thing to remember is that in order to parse a `Zoned`, the interchange format _must_ contain a bracketed time zone identifier like `[America/New_York]` or even `[-04:00]`. Otherwise, you can parse pretty much anything with a date into a `jiff::civil::DateTime`. For example, _all_ of the above examples will parse into a `jiff::civil::DateTime`.

Reply:
Let me know if this doesn't quite answer things for you. (I've only had a quick glance at the code below.)

Reply:
At the risk of digressing completely... let me describe our use case. I think it might be something that other `jiff` users also need. But I'm not sure.

I want to provide a way for Numbat users (most of which will not have a detailed knowledge about timezones and UTC offsets) to enter datetimes as a string *without having to provide a format string*. I'm later planning to add strptime-functionality (i.e. *with* a format string) as well. But for convenience reasons, I want to provide a function that simply takes a string and returns a (zoned) date-time.

We provide a long list of supported formats in our [documentation](https://numbat.dev/doc/date-and-time.html#date-time-formats). But ideally, I want users to have something that *just works*. That means I want to support a large range of popular formats.

In Numbat, we don't provide all layers of safety that jiff provides. For example: we currently don't support datetimes *without* timezones. We do this by providing fallbacks/defaults: If a user does not supply a timezone (or offset) in the input string, we'll use their *local* timezone. If that is not available either, we'll assume UTC. I understand that this might sound like heresy to someone who just implemented a datetime library, but so far, this approach works fine for Numbat.

Now the problem I have is this: I don't know upfront if I am going to parse a `jiff::civil::DateTime` or a `jiff::Zoned`. So what I'm doing now is to try and parse into a `jiff::Zoned` at first (with formats that provide timezone information). And if that doesn't work, try to parse into a `civil::DateTime` (and add timezone information as described above).

And this is why I was a bit hesitant to use a "black box" like `jiff::fmt::temporal` at first. But now that I think about it some more, I guess I can use it in the same way that I do for the other strptime-based parsers. I need to try to parse into `jiff::Zoned` at first, and then fall back to parsing into a `civil::DateTime` afterwards.

The only thing that would be bad is if somehow, temporal would successfully parse something into a `DateTime` that could otherwise be parsed into a `Zoned`. But even then, I could try *all* parsers with `Zoned` at first, and only then attempt to parse `civil::DateTime`s.

Sorry for the long (and probably not every useful) answer. As you can see, I needed to sort my own thoughts at first. I don't really know if there is anything that `jiff` can help with here. Having even more supported formats than just RFC 3339 should be a good thing for my use case.

Reply:
Yeah I think you are on the right track here.

In terms of "provide a datetime and have it just work," I think the main problem there is that it's getting very close to wanting locale support, which is something I've tried hard to _not_ solve in Jiff. (With strtime actually being the singular exception to that because it is just so useful.)

I am open to supporting something like this where you just try to parse anything and make sense of it, probably with some assumptions. But I'm not yet sure what that would look like and whether it would be free of footguns.

But yeah, in a general datetime library, "just assume local time if no TZ/offset is present" is a rather large footgun.